compiler/rustc_typeck/src/structured_errors/wrong_number_of_generic_args.rs

```diff
@@ -626,32 +626,42 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
         let remove_entire_generics = num_redundant_args >= self.gen_args.args.len();
 
         let remove_lifetime_args = |err: &mut DiagnosticBuilder<'_>| {
-            let idx_first_redundant_lt_args = self.num_expected_lifetime_args();
+            let mut lt_arg_indices = Vec::new();
+            let mut found_redundant = false;
+            for (i, arg) in self.gen_args.args.iter().enumerate() {
+                if let hir::GenericArg::Lifetime(_) = arg {
+                    if i > self.num_expected_lifetime_args() - 1 {
+                        found_redundant = true;
+                    }
+                    lt_arg_indices.push(i);
```

Comment:
```suggestion
                    lt_arg_spans.push(arg.span());
```
Spans are the only thing for which `lt_arg_indices` is used, so we can just collect spans.

Reply:
Same applies to the logic for type/const parameters below.